src/index.js

```diff
@@ -31,11 +37,69 @@ const getCandidates = function (el, includeContainer, filter) {
   return candidates;
 };
 
+const getCandidatesIteratively = function (
+  elements,
+  includeContainer,
+  options
+) {
```

Comment:
For future maintenance, it would be really helpful to have a JSDoc comment on this function that details what the parameters are -- especially `options` and all possible properties. It's a bit too hard to decipher it easily just from reading the code.

Reply:
That should also include details about `getShadowRoot()` option and its possible return values, since it looks like it can return probably either a boolean or an Element?

Reply:
will add jsdocs.

No sure `getShadowRoot` full description belongs here, as it is used in other flows as well, and is part of the external api, I will find a place for it that will hopefully be clear.